src/models/FlowRunHistoryMap.ts

```diff
@@ -0,0 +1,23 @@
+import { RunHistory } from '@/models/RunHistory'
+import { StateHistory } from '@/models/StateHistory'
+import { mapper } from '@/services/Mapper'
+import { DateString } from '@/types/dates'
+import { dateFunctions } from '@/utilities/timezone'
+
+export class FlowRunHistoryMap {
```

Comment:
This is cool. Maybe something we could make generic in the future. A composition that accepts a reactive array and a method for how to group it that returns a `reactive<Map>`